akamai/cache-client.go

```diff
@@ -254,9 +254,10 @@ func (cpc *CachePurgeClient) purgeBatch(urls []string) error {
 
 		err := cpc.purge(urls)
 		if err != nil {
-			if _, ok := err.(errFatal); ok {
+			var errorFatal errFatal
+			if errors.As(err, &errorFatal) {
 				cpc.purges.WithLabelValues("fatal failure").Inc()
-				return err
+				return errorFatal
```

Comment:
This should return `err`. If `err` was a wrapper around the inner `errorFatal`, returning just `errorFatal` would throw away useful context.

We happen to know that's not the case here, but in general it's good and correct to return the original error, unless there's a specific reason to do otherwise.